### ISSUE TITLE
Spotify Premium comes back but gets fixed after app restart

### ISSUE THREAD

**Issue by wqffi** [whoeevee/EeveeSpotify#18]
Really weird bug where if you swipe out of Spotify for a bit and come back to it, premium will be lost but once you fully close and reopen the app it has premium again. Wondering if this can be fixed

Swift 1.2, iOS 17.4.1 iPhone 15 Pro Max

**Comment by asdfzxcvbn** [comment]
yeah, this shouldn't happen too much, but it will happen on occasion. it's been very stable for me, actually. has this happened to you more than once in the span of a day or so?

**Comment by timfea926** [comment]
I only noticed it happening once and then turned it off bc I hate getting ads on Spotify.

**Comment by wqffi** [comment]
Yeah happens everytime I leave Spotify open in the background. When stuff is playing it's perfectly fine but if I'm not playing music and swipe out of it and come back after a few seconds, a restart is needed

**Comment by asdfzxcvbn** [comment]
maybe try turning off background refresh? mine works fine for hours on end with it off.

**Comment by wqffi** [comment]
Let me try

**Comment by wqffi** [comment]
> maybe try turning off background refresh? mine works fine for hours on end with it off.

That seemed like it works. Might be worth adding into notes and I installed via Signulous. Would be cool maybe if there is a way to keep background refresh on and keep it working. (I mean it works regardless just needed a simple app restart)

**Comment by asdfzxcvbn** [comment]
why would you prefer background app refresh on?

**Comment by wqffi** [comment]
> why would you prefer background app refresh on?

I personally don't know and not sure if there's a specific functionality that could be nice for Spotify. Some apps really depend on it others don't, Spotify probably being one that doesn't need it. It is just nice to have a app that works in all modes ;) 

But hey I applaud this heavily for a new Spotify version that works unlike old spotilife

**Comment by asdfzxcvbn** [comment]
it just wastes your battery. i personally have it off for all apps systemwide and nothing is broken. i will add this to the readme, though, thanks
